src/trajectory/util.cpp

```diff
@@ -308,5 +309,72 @@ UniqueInterpolatedPtr toR1JointTrajectory(const Interpolated& trajectory)
   return rTrajectory;
 }
 
+//==============================================================================
+void saveTrajectory(const Spline& trajectory, const std::string& savePath)
+{
+  std::ofstream file(savePath);
+  Eigen::IOFormat cleanFmt(
+      Eigen::StreamPrecision, Eigen::DontAlignCols, ",", ",", "", "", "[", "]");
+
+  statespace::ConstStateSpacePtr trajSpace = trajectory.getStateSpace();
+
+  file << "start_time: " << trajectory.getStartTime() << std::endl;
+  file << "dofs: " << trajSpace->getDimension() << std::endl;
+  file << "traj_order: " << trajectory.getNumDerivatives() << std::endl;
+  file << "num_segments: " << trajectory.getNumSegments() << std::endl;
+  Eigen::VectorXd position(trajSpace->getDimension());
+
+  for (std::size_t i = 0; i < trajectory.getNumSegments(); ++i)
+  {
+    file << "seg_" << i << ":\n";
+    auto startState = trajectory.getSegmentStartState(i);
+    double duration = trajectory.getSegmentDuration(i);
+    auto segmentCoeff = trajectory.getSegmentCoefficients(i);
+
+    // Convert start state to position
+    trajSpace->logMap(startState, position);
+
+    file << "  coefficients: " << segmentCoeff.format(cleanFmt) << std::endl;
+    file << "  duration: " << duration << std::endl;
+    file << "  start_state: " << position.format(cleanFmt) << std::endl;
+  }
+}
+
+//==============================================================================
+UniqueSplinePtr loadSplineTrajectory(
+    const std::string& trajPath, const MetaSkeletonStateSpacePtr& stateSpace)
+{
+  YAML::Node trajFile = YAML::LoadFile(trajPath);
+  double startTime = trajFile["start_time"].as<double>();
+  std::size_t dofs = trajFile["dofs"].as<std::size_t>();
+  std::size_t numCoeffs = trajFile["traj_order"].as<std::size_t>() + 1;
+  std::size_t numSegments = trajFile["num_segments"].as<std::size_t>();
+
+  auto Trajectory
+      = ::aikido::common::make_unique<Spline>(stateSpace, startTime);
+
+  for (std::size_t i = 0; i < numSegments; ++i)
+  {
+    auto segment_id = "seg_" + std::to_string(i);
+    Eigen::MatrixXd coeffVec
+        = trajFile[segment_id]["coefficients"].as<Eigen::MatrixXd>();
```

Comment:
Why `MatrixXd` rather than `VectorXd`? It seems to just be a vector in the yaml file?

Reply:
No particular reason, while implementing I may have associated `MatrixXd` with all coefficient related variables I used. But yes, having this as `VectorXd` definitely makes more sense.